src/pages/index.js

```diff
@@ -45,40 +89,106 @@ addCardMOB.addEventListener("click", () => {
   addPlacePopup.open();
 });
 
+profilePic.addEventListener("click", () => {
+  profilePicPopup.open();
+});
+
 //***************************************************************************************************************************//
 //                                             Event Handlers                                                                //
 //***************************************************************************************************************************//
 
 function handleEditProfile(userData) {
+  const submitButton = editProfileModal.querySelector(
+    "#edit-profile-modal-submit-button"
+  );
+  submitButton.textContent = "Saving...";
+  api.patchProfileInfo(userData.name, userData.job);
   userInfo.setUserInfo(userData);
   editProfilePopup.popupForm.reset();
   editProfileFormValidator.toggleButtonState();
   editProfilePopup.close();
```

Comment:
This should be in block `then`

You need to **reset** the form only  in block `then` not to do that immediately after `submit` because when there is an error the popup will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more